src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -704,7 +704,7 @@ private void HandleSemanticTokens<T>(string id, T semanticTokensParams, string m
         /// <param name="queryParams">Collection of query params.</param>
         private void PublishControlTokenNotification(ControlTokens controlTokensObj, NameValueCollection queryParams)
         {
-            if (controlTokensObj == null || controlTokensObj.Size() == 0 || queryParams == null)
```

Comment:
We could potentially remove the Size method from ControlTokens, since it is not used anymore

Reply:
Also we are initializing controlTokensObj to null and this check here would also block the sending for empty control tokens. Might have to remove the check

Reply:
It is used in your test so didn't remove it. Wanted to but your test needs it so let's keep. No it won't it is only null when we do range semantic tokens. It is never null when we do full semantic tokens. We only do control highlighting in full semantic tokens